source/common/router/config_impl.cc

```diff
@@ -400,9 +402,14 @@ std::string RouteEntryImplBase::newPath(const Http::HeaderMap& headers) const {
     final_path = headers.Path()->value().c_str();
   }
 
-  ASSERT(headers.ForwardedProto());
-  return fmt::format("{}://{}{}", headers.ForwardedProto()->value().c_str(), final_host,
-                     final_path);
+  if (https_redirect_) {
+    final_scheme = "https";
```

Comment:
Some notes here (which I hope you'll see since GitHub has hidden the thread):
* I did make this change
* My C++ is pretty rusty, but I'm not sure why this method uses `char*` rather than just `std::string`. Could save a bunch of `.c_str()`s?
* There is [another place](https://github.com/envoyproxy/envoy/blob/4d689efe5e1dc7144ef04dc1a410e96087c2be3f/source/common/router/config_impl.cc#L730-L749) in this file where the `"https"` string literal is used.